### DISCUSSION TITLE
BRouter 1.7

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#149] [Show and tell]
BRouter 1.7 is available:

https://play.google.com/store/apps/details?id=btools.routingapp

Release notes:
- https://github.com/abrensch/brouter/releases/tag/v1.7.0
- https://github.com/abrensch/brouter/releases/tag/v1.7.1

BRouter 1.7 offers the "**Straight line**" option in route planning.

**Comment by 0709wiwiwi** [comment]
BRouter 1.7.0
NEW by Tool:  **CorrectMisplacedViaPoints**.
The correction range in m is freely adjustable.
To activate add next into the different profiles.brf 

assign correctMisplacedViaPoints = 1
assign correctMisplacedViaPointsDistance = 40

![image](https://user-images.githubusercontent.com/54167201/235419234-c16ad2ac-7ec3-494e-ad47-429c45d50a8f.png)


See the practical effect by set distance range = 40m.
https://youtu.be/85Tkf4T1JS0

**Comment by WolfgangZ42** [comment]
I tested it two weeks ago with this .kurviger file: [https://kurv.gr/rDnm2](https://kurv.gr/rDnm2) and it works perfekt, as you may see here:

https://user-images.githubusercontent.com/121806238/235436738-e10c8177-35f6-4cde-b3d2-d00b1d5fcd87.mp4

I reported it already but can't find it in all this github stuff ...

**Comment by MotoUKRider** [comment]
For anyone wishing to update their BRouter and who feeds GPX files into Cruiser (from BaseCamp), the first Route I tried wouldn't "route" using the default/latest `"car-fast"` profile or my own slightly modified (less barriers/gates) one (they appear identical bar my modifications).

Interestingly, a rather complex route in Scotland worked fine, yet a simple loop out to the coast and back caused an issue.  The online BRouter (both within app and Desktop Cruiser) work fine.   I'll use my backup phone to test this new version with my Routes and report back.

**Annoyance:** I had to roll back to one of the earlier beta versions AND re-download the map tiles because, even when you opt to keep the data during uninstall, the older APK won't install if there's a newer version (or data) present.  That is *really* annoying - GoogleMaps does a similar thing where if you update the Maps app or Google itself, your offline maps stop working (despite lasting a year).

**Reply by devemux86** [reply]
> test this new version with my Routes and report back

@MotoUKRider thanks for the tests!

I do not know how often @afischerdev or other BRouter developers read the Cruiser discussions.
It would be useful if anyone finds something strange in BRouter 1.7 to also report it to the [BRouter repository](https://github.com/abrensch/brouter/).

**Reply by MotoUKRider** [reply]
Yes, I probably will - I have just tried a few things and it boils down to roundabouts and slip-roads as starting points - replicated in several places on a map using duplicate "preferences" on both devices.  If you attempt to start from a roundabout (ie: a place where fuel/services may be) then it won't "route".  The on/off-ramp/slip-road does not route, either.  You have to start the Route on the main carriageway or a little back from the roundabout.

See example below:
It will "route" on the main A15 towards the bridge (Humber Bridge), but not on the entry-ramp, so you have to fine a "clear" piece of road.

![cruiser-brouter-issue](https://user-images.githubusercontent.com/125696109/235621409-e78845c3-0bbc-4d13-a0be-5574933720ef.jpg)

Given that this issue would affect any user of BRouter+Cruiser, certainly worth posting here, too :)

**Reply by devemux86** [reply]
BRouter appears to be more strict with waypoint placement compared to other routers.
It seems to like the waypoints to be closer to the road network?

Is there any parameter for "snap to street" in the BRouter profiles that we can change?

**Reply by MotoUKRider** [reply]
The odd thing is that the previous "Play" version was fine - and a further test suggests that it's only the *Start* point causing a problem.  A Route with a Shaping-Point in the same place was able to route correctly.  I'll add this to my "issue" over on the BRouter page!

**Reply by 0709wiwiwi** [reply]
https://github.com/abrensch/brouter/issues/519#issuecomment-1489805580
Whether this has anything to do with it I don't know. Thoroughly tested in a penultimate Beta version this is not. Those last ultimate changes are quite often surprising ;-)

Unfortunately, also app downgrades still need to be prepared very carefully.
https://github.com/devemux86/cruiser/discussions/80#discussioncomment-5450318
So that experience was disappointing. I had expected it to be comfortable. Not so, on the contrary.

**Reply by MotoUKRider** [reply]
Hah!  I wouldn't mind the "bug" so much if was easy to roll-back without losing data - It's not very efficient to require users to re-download data en-masse when it can just be (re)stored next time.  That's also a problem with apps within the Play Store, unless you're savvy enough to take an APK backup - you update to the latest version and find a problem, you can often find yourself.....errrr, stuck!

From now on, the spare phone gets the updates and if it doesn't "break", I'll chance it on the main one :)

**Comment by devemux86** [comment]
BRouter is already progressing rapidly with new tags in its database
that will allow us to calculate **offline** routes considering towns, traffic, noise, forests, rivers, etc.

Unfortunately there is currently an issue with _new installs_ of the latest BRouter app.
BRouter app retrieves updated profiles while the database is not yet updated, which leads to failed routes.

I reported it in BRouter repository: https://github.com/abrensch/brouter/issues/580

Currently, the problem exists with the Bike and Walk / Hike profiles.
Car and Motorcycle profiles work correctly as they use Cruiser's custom _drving-cruiser_ profile.

It is safer for users to have Cruiser integrate BRouter's original profiles, so we can avoid such problems in the future.
They can be available in the next Cruiser update, together with Cruiser's custom _driving-cruiser_ profile.

**Reply by devemux86** [reply]
Quick fix until the next Cruiser update and if you happen to see this BRouter issue:

- Go to BRouter profiles folder: `/Android/media/btools.routingapp/brouter/profiles2/`
- Rename the file `lookups.dat.tmp` to `lookups.dat` overwriting the old file

**Reply by devemux86** [reply]
This will also be fixed from the BRouter side in the next BRouter app update.

**Reply by devemux86** [reply]
BRouter profiles are now included in [Cruiser 3.0.25](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

Along with the custom Cruiser driving profile.

**Comment by devemux86** [comment]
BRouter 1.7.1 is available:
- https://github.com/abrensch/brouter/releases/tag/v1.7.1

Besides the many improvements and great support from @afischerdev, very important is:
- Update matching points rules on areas with longer distances between way points

Which will improve routing in all those cases where waypoints could not be matched with nearby roads.

**Reply by DolomitesTom** [reply]
What is the installation instruction for this? Or how long does it take until it is available via play store?

**Reply by DolomitesTom** [reply]
Just seen that it is available in play store, perfect, thanks to the developer.
